sys/fmt/fmt.c

```diff
@@ -183,12 +183,12 @@ size_t fmt_u16_dec(char *out, uint16_t val)
 
 size_t fmt_s32_dec(char *out, int32_t val)
 {
-    int negative = (val < 0);
+    size_t negative = (val < 0 ? 1 : 0);
```

Comment:
why not unsigned, as used in other places? Even (unsigned) char would do, but will likely not be more optimal in size, anyway.